src/libraries/System.Console/src/System/ConsolePal.iOS.cs

```diff
@@ -0,0 +1,188 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+
+namespace System
+{
+    internal sealed class NSLogStream : Stream
+    {
+        public override void Flush() { }
+
+        public override int Read(byte[] buffer, int offset, int count) => throw new NotSupportedException();
+
+        public override long Seek(long offset, SeekOrigin origin) => throw new NotSupportedException();
+
+        public override void SetLength(long value) => throw new NotSupportedException();
+
+        public override unsafe void Write(byte[] buffer, int offset, int count)
+        {
+            if (count > 0 && buffer.Length >= offset + count)
+            {
+                fixed (byte* ptr = buffer)
+                {
+                    Interop.Sys.Log((IntPtr)(ptr + offset), count);
+                }
+            }
+        }
+
+        public override bool CanRead => false;
+
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => true;
+
+        public override long Length => throw new NotSupportedException();
+
+        public override long Position
+        {
+            get => throw new NotSupportedException();
+            set => throw new NotSupportedException();
+        }
+    }
+
+    internal static class ConsolePal
+    {
+        public static Stream OpenStandardInput() => throw new PlatformNotSupportedException();
+
+        public static Stream OpenStandardOutput() => new NSLogStream();
+
+        public static Stream OpenStandardError() => new NSLogStream();
+
+        public static Encoding InputEncoding => throw new PlatformNotSupportedException();
+
+        public static void SetConsoleInputEncoding(Encoding enc) => throw new PlatformNotSupportedException();
+
+        public static Encoding OutputEncoding => Encoding.Unicode;
+
+        public static void SetConsoleOutputEncoding(Encoding enc) => throw new PlatformNotSupportedException();
+
+        public static bool IsInputRedirectedCore() => throw new PlatformNotSupportedException();
+
+        public static bool IsOutputRedirectedCore() => throw new PlatformNotSupportedException();
+
+        public static bool IsErrorRedirectedCore() => throw new PlatformNotSupportedException();
```

Comment:
I'm wondering if some of these would be better as just returning a hardcoded value.  Can stdin/out/err be redirected?  If not, then these could just return false.

Reply:
makes sense, fixed, thanks.